docs/new-architecture-backward-turbomodule.md

```diff
@@ -0,0 +1,709 @@
+
+For sake of simplicity, we will go on with this manual setup, adding files while we work.
+Remember to update the `<Your Name>`, `<your_email@your_provide.com>`, and all the `<your_github_handle>` tokens, if you want to follow this approach with your information.
+
+### Create a Testing Environment
+
+To develop a Native Module we would like to test the JS side and to develop safely the native parts. A library usually does not come with a testing aplication out-of-the-box. So, we are going to need one.
+
+:::note
+If you used the React Native CLI to create the module, you have an application called `example` within you `example-library` folder.
+:::
+
+To create a proper test environment, let's navigate to the parent folder of our `example-library` and run the following commands:
+
+```sh
+npx react-native init OldArchitectureApp
+cd OldArchitectureApp
+yarn install
+```
+
+Now, open a new terminal and start metro by running:
+
+```sh
+npx react-native start
+```
+
+Go back to previous terminal and run
+
+```sh
+npx react-native run-ios
+```
+
+When all the processes are done, you should see the default React Native application, running on a simulator.
+We will update the code of this app to test our Native Module.
+
+### Define the JS Shared Code
+
+The next step is to define how the React Native app will use our module. Native modules are registered using the `NativeModule` object provided by the React Native framework.
+
+For this example, we are going to create a manager that asks the GPS authorization to the user.
+
+As first step, let's create a folder named `src` and an `index.js` file into it. Add the following code into the JS file.
+
+```js
+// @flow
+// 1
+import { NativeModules, Platform } from 'react-native';
+
+// 2
+const LINKING_ERROR =
+  'The package "AuthorizationManager" doesn\'t seem to be linked. Make sure: \n\n' +
+  Platform.select({
+    ios: "- You have run 'pod install'\n",
+    default: ''
+  }) +
+  '- You rebuilt the app after installing the package\n' +
+  '- You are not using Expo managed workflow\n';
+
+// 3
+const manager = NativeModules.AuthorizationManager
+  ? NativeModules.AuthorizationManager
+  : new Proxy(
+      {},
+      {
+        get() {
+          throw new Error(LINKING_ERROR);
+        }
+      }
+    );
+
+// 4
+export type AuthorizationStatus =
+  | 'NotDetermined'
+  | 'Restricted'
+  | 'Denied'
+  | 'Authorized'
+  | 'AuthorizedAlways'
+  | 'AuthorizedWhenInUse';
+
+export default manager;
+```
+
+:::info
+This file uses [`Flow`](https://flow.org/) to work with types.
+Types are required for the Turbomodules
+:::
+
+The file imports the `NativeModules` and the `Platform` objects from `react-native` (marked with `// 1` in the code).
+Then, it prepares the manager if the module is properly registered (`// 3`). Otherwise, when we try to use it, we get an error (set up in `// 2`).
+
+Finally, it exposes a type for the possible alternative authorization status and the manager object (`// 4`).
```

Comment:
Can we put those comment directly in the code example? It would be more aligned with the other code examples in the docs.